crypto/err/err.c

```diff
@@ -812,3 +812,26 @@ int ERR_pop_to_mark(void)
     es->err_flags[es->top] &= ~ERR_FLAG_MARK;
     return 1;
 }
+
+int ERR_clear_last_mark(void)
+{
+    ERR_STATE *es;
+    int top;
+
+    es = ERR_get_state();
+    if (es == NULL)
+        return 0;
+
+    top = es->top;
+    while (es->bottom != top
+           && (es->err_flags[es->top] & ERR_FLAG_MARK) == 0) {
+        top -= 1;
+        if (top == -1)
+            top = ERR_NUM_ERRORS - 1;
```

Comment:
Here by the way "fun" question in the context. Common pattern is XXXerr, ERR_add_error_data. Both rely on ERR_get_state to return state or NULL. What if first call to ERR_get_state fails and second succeeds? I suppose what I'm saying is that formally ERR_get_state should not retry if it failed. Or have an extra flag that denotes moment when it's appropriate to retry. Why is it even public? Why did I look? ...